javascript/Spread_operator.md

```diff
@@ -0,0 +1,162 @@
+﻿# Spread Operator in javascript
+
+## Projected time
+
+About 30-40 minutes
+
+- 15 min for spread operator
+- 25 min for examples
+
+## Spread operator
+Spread operator allows an iterable such as strings, array and sets to spread or expand individually inside a receiver.
+
+The **syntax** of spread operator is `var variable  = [...value]`.
+
+## Why it is useful?
++ We can merge two objects together using spread operator.
+Example:
+```
+let obj1 = {a: 1, b: 2, c: 3}
+let obj2 = {d: 4, e: 5, f: 6}
+let merge = {...obj1, ...obj2} // merge is {a: 1, b: 2, c: 3, d: 4, e: 5, f: 6}
+```
++ Spread operator with object is used to make copy of existing object or to make a new object with properties.
+
+## Examples
+
+### Simple object
+
+Given this ```todoApp``` object we can use the object spread syntax and its properties into variables.
+
+```javaScript
+function todoApp(state = initialState, action) {
+ switch (action.type) {
+   case SET_VISIBILITY_FILTER:
+      return { ...state, visibilityFilter: action.filter }
```

Comment:
You should specify the `initialState` object attributes to make them clearly understand how spread operator works or you can change this example as it is clearly an example of reducers in react and that's not what you are trying to explain here. Use some other simple example.